### PR TITLE
Array or List in query string does not get parsed #7712

### PR DESCRIPTION
**Pull request by kishanAnem**
Hi @dougbu
#7712
Added following test cases.

1. PreservesBindingSourceInference_ForFromQueryParameterOnCollectionType()
2. PreservesBindingSourceInference_ForFromQueryOnArrayType()
3. PreservesBindingSourceInference_FromQueryOnArrayTypeWithCustomName()

### CODE REVIEW

**Comment by dougbu** [comment]
@kishanAnem why did you close this PR?

**Comment by kishanAnem** [comment]
Actually it has some conflict. So I closed pr. and trying to open a fresh pr.

**Comment by kishanAnem** [comment]
If it is fine. Please review. let me know correction, I'm very glad to do it. ☺️

File: src/Microsoft.AspNetCore.Mvc.Core/Internal/ApiBehaviorApplicationModelProvider.cs [hunk 1]
```diff
@@ -155,10 +155,9 @@ internal void InferParameterBindingSources(ActionModel actionModel)
                 if (bindingSource == null)
                 {
                     bindingSource = InferBindingSourceForParameter(parameter);
-                    parameter.BindingInfo = new BindingInfo
-                    {
-                        BindingSource = bindingSource,
-                    };
+
+                    parameter.BindingInfo = parameter.BindingInfo ?? new BindingInfo();
```
[the remark this answers is not in the document]
Comment (kishanAnem) [reply src/Microsoft.AspNetCore.Mvc.Core/Internal/ApiBehaviorApplicationModelProvider.cs]: I didn't made any changes there.

**Comment by kishanAnem** [comment]
Will re base my fork and will send pr again. Thank you @dougbu

File: src/Microsoft.AspNetCore.Mvc.Core/Internal/ApiBehaviorApplicationModelProvider.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (kishanAnem) [reply src/Microsoft.AspNetCore.Mvc.Core/Internal/ApiBehaviorApplicationModelProvider.cs]: I think need to rebase forkas said.

File: test/Microsoft.AspNetCore.Mvc.Core.Test/Internal/ApiBehaviorApplicationModelProviderTest.cs [hunk 2]
```diff
@@ -556,6 +558,57 @@ public void PreservesBindingSourceInference_ForFromQueryParameter_WithCustomName
         {
             // Arrange
             var modelMetadataProvider = TestModelMetadataProvider.CreateDefaultProvider();
+=======
```
Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.Core.Test/Internal/ApiBehaviorApplicationModelProviderTest.cs]: @kishanAnem please resolve these conflicts. That is, accept `dev` additions and your additions to reduce the visible changes to what you intended.

Comment (kishanAnem) [reply test/Microsoft.AspNetCore.Mvc.Core.Test/Internal/ApiBehaviorApplicationModelProviderTest.cs]: Sure, working on it. This does not actually commit

**Comment by kishanAnem** [comment]
I think better to close this PR.  I'll create new PR.

**Pull request closed without merging**